dataRetriever/txpool/shardedTxPool.go

```diff
@@ -387,6 +387,11 @@ func (txPool *shardedTxPool) Diagnose(deep bool) {
 	}
 }
 
+// NotifyEviction will be called when a transaction is removed from the main tx pool
+func (txPool *shardedTxPool) NotifyEviction(txHash []byte) {
```

Comment:
Who calls `NotifyEviction`? If called on a large collection, might cause performance issues (since `removeTxFromAllShards` isn't extremely fast (if I remember well).

Reply:
Also, does not seem to be covered by tests.

Reply:
`NotifyEviction` is called on a new go routine, from mx-chain-storage-go, and the calls are protected through a worker pool, allowing max 5 go routines at a time. 

Added tests

Reply:
maybe we can do some benchmarking to have a  clearer view on the performance